docx/blkcntnr.py

```diff
@@ -23,6 +26,15 @@ def __init__(self, element, parent):
         super(BlockItemContainer, self).__init__(parent)
         self._element = element
 
+    def generate_numId(self):
+        """
+        Generate a unique numId value on this container.
+        """
+        while True:
+            numId = random.randint(0, 999999)
```

Comment:
Instead of random numbers, why not just
```
from itertools import count
...
for numId in count(len(self._element.xpath('//w:numId')):
    if not self._element.xpath('//w:numId[@w="{}"]'.format(numId):
        return numId
```